lib/biran/erb_config.rb

```diff
@@ -10,11 +12,12 @@ def initialize(config, name, extension, source, output_dir, output_name)
       @source_dir = source
       @output_dir = output_dir
       @output_name = output_name
+      @template_contents = process_erb
```

Comment:
This holds the template contents as an instance variable for successive runs instead of having to reload each time.

Reply:
I think this is the only part that makes me hesitant. I think the order is important and not sure how to enforce for later users. process_erb requires some of these other instance variables to exist. So not sure if it can be moved to a method that runs "after" initialize by naming it so its clear it should be last in this list or a simple comment line would help...

Reply:
How about adding explicit error raising inside `process_erb` if this vars are nil?

like `raise 'Missing @output_dir' if @output_dir.nil?`

Reply:
So adding validation in the use of the template variable for the required items, we ensure that future users would see that right away and resolve the order issue.
That could work. I just didn't know if there was any other way to call this from initialize but after initialize was complete... like chain the action.
Would it be better to write a test that enforces the order? That way when the test breaks, you would see you did something wrong as a developer?   Or both??

Reply:
Before initalize nothing happens, so we can't build it there... you can add a `before_process_erb` action block, and add any validations in there. like:

```
before_process_erb do
  step_1
  step_2
  step_3
end
```

and had a method such as:

```
def before_process_erb
  yield
  raise 'Missing step' unless @step1 and @step2 and @setp3
  process_erb
end
```

that I think communicate best the flow.